WordPress/src/main/java/org/wordpress/android/ui/reader/usecases/ReaderSiteFollowUseCase.kt

```diff
@@ -28,44 +30,56 @@ class ReaderSiteFollowUseCase @Inject constructor(
 ) {
     private var continuation: Continuation<Boolean>? = null
 
-    suspend fun toggleFollow(post: ReaderPost) = flow<FollowSiteState> {
+    suspend fun toggleFollow(param: Param) = flow<FollowSiteState> {
         if (!networkUtilsWrapper.isNetworkAvailable()) {
             emit(NoNetwork)
         } else {
-            val isAskingToFollow = !readerPostTableWrapper.isPostFollowed(post)
-            val showEnableNotification = !readerUtilsWrapper.isExternalFeed(post.blogId, post.feedId) &&
-                    !post.isFollowedByCurrentUser
-            emit(PostFollowStatusChanged(post.blogId, isAskingToFollow, showEnableNotification))
-
-            performAction(post, isAskingToFollow)
+            val isAskingToFollow = when (param) {
+                is FromPost -> !readerPostTableWrapper.isPostFollowed(param.post)
+                is RecommendedSite -> param.isAskingToFollow
+            }
```

Comment:
I wonder if we could simplify it to:

```suggestion
            val isAskingToFollow = readerBlogTableWrapper.isSiteFollowed(param.blogId, param.siteId)
```